solutions/advanced/cf-891C

```diff
@@ -0,0 +1,150 @@
+---
+id: cf-891C
+source: CF
+title: Envy
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/55841)
+
+## Explanation
+
+We can solve this using some MST intuition. In Kruskal's algorithm, we are to sort the edges by non-decreasing weight, and try uniting vertices if they aren't already in the same component. Here, we can try to check every query using this algorithm, by processing each edge of each query offline.
+
+As we consider edges from non-decreasing weight, we can keep track of a partially-built MST. An edge can only be in a MST if we processed all edges that have weights less than this edge, and we are able to unite the endpoints of the edge. This is because edges that have the same weight are arbitrary, so if we prioritize considering this edge then another MST with the same total weight is constructed.
+
+Thus, for each query, we need all edges to satisfy this condition. We can process all edges in non-decreasing weight, and for each query, try to unite the endpoint of all the edges that have this weight. If any edge does not satisfy, the whole query fails.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(q \log n)$
```

Comment:
Update the complexity to the correct value

Reply:
is it not correct?

Reply:
There has to be a factor of $m\log m$ in there somewhere for the sorting right?

Reply:
wdym i didnt sort nothing

Reply:
Ok but you read in m + q numbers so that should definitely be in there?

Reply:
bruh fine